liste.c

```diff
@@ -0,0 +1,228 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* define list struct */
+typedef struct node
+{
+	int value;
+	int print_opt;
+	struct node *next;
+}Node;
+
+
+/* defines how the element will be displayed */
+void print_n(int val, int n_opt)
+{
+	switch( n_opt )
+	{
+	case 1: printf("Node value is: %d\n", val);
+				break;
+	case 2:
+			printf("Node value is: 0x%xh\n", val); 
+				break;
+	default:
+				break;
+	}
+} 
+
+
+/* add a new node to the end of the list */
+Node* add_node(Node** up_node, int val, int print_option)     
+{
+	/* memory allocation for the new node */
+	Node* new_node = (Node*)malloc(sizeof(Node));
+
+	/* copy the given data to new node fields */
+	new_node -> value = val;
+	new_node -> print_opt = print_option;
+	new_node -> next = NULL;
+
+	Node* nnp = *up_node;
+
+	/* check if the list is empty */
+	if(*up_node == NULL)
+	{
+		*up_node = new_node;
+
+	}
+	else
+	{
+		/* search for the last node of the given list
+		   and then insert the new node */
+		while(nnp->next != NULL)
+		{
+			/* go to the next node */
+			nnp = nnp->next;
+		}
+
+		nnp->next = new_node;
+	}
+
+	return new_node;
+}
+
+/* print all the list */
+void print_node(Node* head)
+{
+	Node* my_node = head;
+
+	while(my_node != NULL)
+	{
+		/* call function to print node values */
+		print_n(my_node->value, my_node->print_opt);
+
+		/* go to the next node */
+		my_node = my_node->next;
+	}		
+}
+
+
+/* delete the first element of the list if matches*/
+Node* delete_first(Node* head, int val)
```

Comment:
My suggestion is to create a function to delete the first occurence of a value in a list and then call it consecutively until all the nodes  with that values are deleted.

delete_first - delete the first occurence of val in the list
delete_node - delete all the occurences of val in the list by using the delete_first